### PR TITLE
feat: map chain id in DID

### PR DESCRIPTION
**Pull request by Harasz**

### CODE REVIEW

**Comment by Harasz** [comment]
@JGiter @jrhender @nichonien I want you all guys to take a look into this PR and let me know if I am not missing any part regarding new format of the DID.

**Review by JGiter — APPROVED** [review]
Although we can process role definition here, I see DomainReader as responsible only for reading. They way how blockchain entities persisted and used should be defined in cache server

File: src/DomainReader.ts [hunk 1]
```diff
@@ -203,7 +203,12 @@ export class DomainReader {
         return textProps;
       }
       if (DomainReader.isRoleDefinition(textProps)) {
-        return await this.readRoleDefResolver_v1(node, textProps, ensResolver);
+        return await this.readRoleDefResolver_v1(
+          node,
+          textProps,
+          ensResolver,
+          resolverAddress,
```
Comment (jrhender) [inline src/DomainReader.ts:210]:
```suggestion
```

File: src/DomainReader.ts [hunk 2]
```diff
@@ -236,7 +241,12 @@ export class DomainReader {
         return textProps;
       }
       if (DomainReader.isRoleDefinition(textProps)) {
-        return await this.readRoleDefResolver_v2(node, textProps, ensResolver);
+        return await this.readRoleDefResolver_v2(
+          node,
+          textProps,
+          ensResolver,
+          resolverAddress,
```
Comment (jrhender) [inline src/DomainReader.ts:248]:
```suggestion
```

**Review by jrhender — CHANGES_REQUESTED** [review]
A really great start, but I think this can be improved. Writing a PR for your PR now 😄

**Comment by jrhender** [comment]
@Harasz Please see https://github.com/energywebfoundation/iam-contracts/pull/71

File: src/DomainReader.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ import { RoleDefinitionResolverV2__factory } from "../ethers/factories/RoleDefin
 import { ResolverContractType } from "./types/ResolverContractType";
 import { ERROR_MESSAGES } from "./types/ErrorMessages";
 import { ENSRegistry } from "../ethers/ENSRegistry";
+import { EWC_CHAIN_ID } from ".";
```
Comment (jrhender) [inline src/DomainReader.ts:28]: A bit weird that this `import` is separate from the `VOLTA_CHAIN_ID` import. Could it be put with them?

**Review by jrhender — APPROVED** [review]
A small comment, but looks great, thanks!

**Pull request merged**